src/mixer/basetrackplayer.cpp

```diff
@@ -650,6 +666,34 @@ void BaseTrackPlayerImpl::slotCloneChannel(EngineChannel* pChannel) {
     slotLoadTrack(pTrack, false);
 }
 
+void BaseTrackPlayerImpl::slotLoadTrackFromDeck(double d) {
+    int deck = static_cast<int>(d);
+    if (deck >= 1) {
+        slotLoadTrackFromGroup(PlayerManager::groupForDeck(deck - 1));
+    }
+}
+
+void BaseTrackPlayerImpl::slotLoadTrackFromSampler(double d) {
+    int sampler = static_cast<int>(d);
+    if (sampler >= 1) {
+        slotLoadTrackFromGroup(PlayerManager::groupForDeck(sampler - 1));
+    }
+}
+
+void BaseTrackPlayerImpl::slotLoadTrackFromGroup(const QString& group) {
+    EngineChannel* pChannel = m_pEngineMaster->getChannel(group);
+    if (!pChannel) {
+        return;
+    }
+
+    TrackPointer pTrack = pChannel->getEngineBuffer()->getLoadedTrack();
```

Comment:
A null check for pChannel->getEngineBuffer() can't hurt here.

Reply:
The `BaseTrackPlayerImpl` constructor creates a new `EngineDeck` which creates the `EngineBuffer`, and not even there is a a check for `pEngineBuffer`, so why here? if it was missing for whatever reason Mixxx would have fallen apart long before LoadTrackFromDeck could be triggered.

Reply:
Yes, right. So this is a perfect use case of 'gsl::not_null<>. But probably a topic for an extra pull request.